src/build.h

```diff
@@ -35,6 +35,42 @@ struct Node;
 struct State;
 struct Status;
 
+
+// Set of ready edges, sorted by priority
+class EdgeQueue {
+  struct EdgePriorityCompare {
+    bool operator()(const Edge* e1, const Edge* e2) const;
+  };
+
+  std::priority_queue<Edge*, std::vector<Edge*>, EdgePriorityCompare> queue_;
+
+ public:
+  void push(Edge* edge) {
+    queue_.push(edge);
+  }
+
+  template <typename It>
+  void push(It first, It last) {
+    for (; first != last; ++first) {
+      push(*first);
+    }
+  }
+
+  Edge* pop() {
+    Edge* ret = queue_.top();
+    queue_.pop();
+    return ret;
+  }
+
+  void clear() {
+    queue_ = std::priority_queue<Edge*, std::vector<Edge*>, EdgePriorityCompare>();
+  }
```

Comment:
Note that a simpler version to get the same result is to subclass std::priority_queue which exposes the container as a protected member named `c` as in:

```
class EdgeQueue : public std::priority_queue<Edge*, std:;vector<Edge*>, EdgePriorityCompare> {
 public:
  void clear() { c.clear(); }
}
```